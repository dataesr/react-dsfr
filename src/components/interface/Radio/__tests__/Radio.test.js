import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { v4 as uuidv4 } from 'uuid';
import { Radio } from '../index';

jest.mock('uuid', () => ({
  v4: jest.fn(),
}));

describe('<Radio />', () => {
  beforeEach(() => {
    uuidv4.mockImplementationOnce(() => 'xxxxxxx');
  });
  it('renders correctly', () => {
    const component = renderer
      .create(
        <Radio
          label="Label de la radio"
          value="value"
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('in document', () => {
    render(
      <Radio
        label="Label de la radio"
        value="value"
      />,
    );
    const search = screen.getByTestId('radio-testid');
    expect(search).toBeInTheDocument();
  });

  // test de la prÃ©sence de la class rf-radio-group
  it('has rf-radio-group class', () => {
    const { container } = render(
      <Radio
        label="Label de la radio"
        value="value"
      />,
    );
    expect(container.getElementsByClassName('rf-radio-group').length).toBe(1);
  });

  // test de la class d'erreur
  it('has rf-select class', () => {
    const { container } = render(
      <Radio
        label="Label de la radio"
        value="value"
        messageType="error"
      />,
    );
    expect(container.getElementsByClassName('rf-radio-group--error').length).toBe(1);
  });
});
