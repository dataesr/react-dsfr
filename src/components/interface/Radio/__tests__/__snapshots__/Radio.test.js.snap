// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> renders correctly 1`] = `
<div
  className="rf-radio-group"
>
  <input
    data-testid="radio-testid"
    id="xxxxxxx"
    name="radio"
    onChange={[Function]}
    type="radio"
    value="value"
  />
  <label
    className="rf-label"
    htmlFor="xxxxxxx"
  >
    Label de la radio
  </label>
  
  
</div>
`;
